### PR TITLE
feat: dexie powered message db

### PR DESCRIPTION
**Pull request by neekolas**
### <samp>🤖 Generated by Copilot at 2500b9b</samp>

This pull request refactors several components and hooks in the `./apps/web/src/components` directory to use the local database for profile and preview message data instead of the global store. It also updates some dependencies in `pnpm-lock.yaml` and swaps the order of some class names for consistency. Additionally, it adds the `useMessageDb` hook and the `message-db` module to handle the local database operations using the Dexie library.

When a user has been away for a long time, we still need to check if a new message has been sent in any of their conversations. This process is much faster than before. That's because we can check only for new messages that came in after the last message in the DB. For most conversations, that will return no results, saving data transfer and avoiding an expensive decryption.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Enhancement (non-breaking small changes to existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### <samp>🤖 Generated by Copilot at 2500b9b</samp>

*  Use the Dexie library and its React hooks to create and access a local database for profile and preview message data ([link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-5648372a44a6312ee755a3755b67eae25eda61d21826c43f088f8594b26db0d7R1-R116), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-c555a2deee77a53c6da774d69866edba991f7fbf97d2d8d4cff9bc1203146cc7R1-R42), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b971652b1e5c5c7d512e6c7091d49059c858f7e50f84d8364cd66384c14bda4aR9-R10), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b4f3b185340d536623bfcecce43624e3f8d2155a371a469dacb7aaf372397bf3R15-R16), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-051c5aa8ed716f2a43344cf26ef7f981c4ad1884f3cf442ae394da8e5527aa04R8), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-eda1f4100c8fae29a041adb14d95c174c2313e7056d3922bd0c0b1b77d634778R6))
*  Update the `useMessagePreviews` hook to sync the profile and preview message data from the Lens API and the XMTP client to the local database and update the global store with the decoded preview messages ([link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b4f3b185340d536623bfcecce43624e3f8d2155a371a469dacb7aaf372397bf3L7-R7), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b4f3b185340d536623bfcecce43624e3f8d2155a371a469dacb7aaf372397bf3L22-R30), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b4f3b185340d536623bfcecce43624e3f8d2155a371a469dacb7aaf372397bf3R40-R45), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b4f3b185340d536623bfcecce43624e3f8d2155a371a469dacb7aaf372397bf3L50-R85), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b4f3b185340d536623bfcecce43624e3f8d2155a371a469dacb7aaf372397bf3L65-R98), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b4f3b185340d536623bfcecce43624e3f8d2155a371a469dacb7aaf372397bf3R113-R114), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b4f3b185340d536623bfcecce43624e3f8d2155a371a469dacb7aaf372397bf3L88-R124), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b4f3b185340d536623bfcecce43624e3f8d2155a371a469dacb7aaf372397bf3L110-R141), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b4f3b185340d536623bfcecce43624e3f8d2155a371a469dacb7aaf372397bf3L200-R231), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b4f3b185340d536623bfcecce43624e3f8d2155a371a469dacb7aaf372397bf3L218-R249))
*  Update the `useGetMessagePreviews` hook to sync the preview messages from the XMTP client to the local database only once and use the local database data instead of the global store ([link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b971652b1e5c5c7d512e6c7091d49059c858f7e50f84d8364cd66384c14bda4aL26-R31), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b971652b1e5c5c7d512e6c7091d49059c858f7e50f84d8364cd66384c14bda4aL34-R72), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-b971652b1e5c5c7d512e6c7091d49059c858f7e50f84d8364cd66384c14bda4aL72-R78))
*  Update the `Message` component to use the `useGetProfile` hook to get the profile data from the local database instead of the global store ([link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-2ce189b37041c30ccd1e2909ceecbb30ea9b59738f19f0311bf70e40993bca6bR6), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-2ce189b37041c30ccd1e2909ceecbb30ea9b59738f19f0311bf70e40993bca6bL19), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-2ce189b37041c30ccd1e2909ceecbb30ea9b59738f19f0311bf70e40993bca6bL33-R33))
*  Update the `MessageHeader` component to use the `useCallback` hook to wrap the `setFollowing` function and the `useMessageStore` hook to access the `unsyncProfile` action to remove the profile from the synced profiles set when the following status changes ([link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-0b872fbacf4157f72d0c9b94ce54ae519739cf13975d047070f197a3b3fdccbbL7-R8), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-0b872fbacf4157f72d0c9b94ce54ae519739cf13975d047070f197a3b3fdccbbL19-R29), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-0b872fbacf4157f72d0c9b94ce54ae519739cf13975d047070f197a3b3fdccbbL26-R36), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-0b872fbacf4157f72d0c9b94ce54ae519739cf13975d047070f197a3b3fdccbbL42-R56))
*  Update the `PreviewList` and `Details` components to use the `persistProfile` function to save the profile data to the local database and the `setSelectedTab` action to update the selected tab based on the following status of the profile ([link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-eda1f4100c8fae29a041adb14d95c174c2313e7056d3922bd0c0b1b77d634778R21), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-eda1f4100c8fae29a041adb14d95c174c2313e7056d3922bd0c0b1b77d634778L32-R34), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-eda1f4100c8fae29a041adb14d95c174c2313e7056d3922bd0c0b1b77d634778L63-R70), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-051c5aa8ed716f2a43344cf26ef7f981c4ad1884f3cf442ae394da8e5527aa04R30), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-051c5aa8ed716f2a43344cf26ef7f981c4ad1884f3cf442ae394da8e5527aa04L50-R53), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-051c5aa8ed716f2a43344cf26ef7f981c4ad1884f3cf442ae394da8e5527aa04L58-R63))
*  Update the `useMessageStore` hook to remove the unused actions and state for profile data and add the actions and state for tracking the sync status of the preview messages and the profile ids ([link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-bf53ea74161761cc199afb3af1d4962130ed9eb89d191b26bd5fb551e83d534dL5-R8), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-bf53ea74161761cc199afb3af1d4962130ed9eb89d191b26bd5fb551e83d534dL20-R20), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-bf53ea74161761cc199afb3af1d4962130ed9eb89d191b26bd5fb551e83d534dR29-R31), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-bf53ea74161761cc199afb3af1d4962130ed9eb89d191b26bd5fb551e83d534dL63-R65), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-bf53ea74161761cc199afb3af1d4962130ed9eb89d191b26bd5fb551e83d534dL99-R95))
*  Swap the order of the `line-clamp-2` and `mt-2` classes in the `Link` components in the notification components to make the class names consistent ([link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-41c2f9e1cf3ab7e1345e1f746a997efc51556a876cb8965c85236c01f2631d19L14-R14), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-ded014ed30647003114ac4285b5b1609969b714912d6e5163c62eabb757299b8L56-R56), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-5f42170e541d1ad6a9b0da04a30809a1432707bd757ba12b455326bbec3c0af4L64-R64), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-293b8102e069ea62b38ffa4d71d6085030896e2e0a17c540feb44156f099465cL57-R57), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-ad9617ea2e755023ea4fffa277a1bc9a72b1d26641330aa7c7ad441f3e32879bL56-R56))
*  Update the `@xmtp/xmtp-js` dependency version to use the latest version that fixes some bugs and improves performance ([link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL211-R212))
*  Update the `@ethersproject/*` dependency versions to use the latest version that fixes some bugs and improves compatibility ([link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL4024-R4038), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL4052-R4064), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL4076-R4086), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL4096-R4106), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL4116-R4122), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL4128-R4135), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL4142-R4149), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL4158-R4164), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL4170-R4176), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL4182-R4197), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL4212-R4225), [link](https://github.com/lensterxyz/lenster/pull/2711/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL4239-R4256))

## Emoji

🔄🗃️🛠️

### CODE REVIEW

File: apps/web/src/components/Messages/MessageHeader.tsx [hunk 1]
```diff
@@ -16,14 +17,23 @@ interface MessageHeaderProps {
 const MessageHeader: FC<MessageHeaderProps> = ({ profile }) => {
   const router = useRouter();
   const [following, setFollowing] = useState(true);
+  const unsyncProfile = useMessageStore((state) => state.unsyncProfile);
```
Comment (neekolas) [inline apps/web/src/components/Messages/MessageHeader.tsx:20]: I thought I created a bug with my changes, but it looks like the behaviour on prod is the same. Following a user should move them from the `Requested` to the `Following` tab. Now it actually does that.

**Comment by bigint** [comment]
@neekolas seems like some conflicts are there 🙇🏼

File: apps/web/src/components/Notification/Type/LikeNotification.tsx [hunk 2]
```diff
@@ -61,7 +61,7 @@ const LikeNotification: FC<LikeNotificationProps> = ({ notification }) => {
           />
           <Link
             href={`/posts/${notification?.publication?.id}`}
-            className="lt-text-gray-500 linkify mt-2 line-clamp-2"
+            className="lt-text-gray-500 linkify line-clamp-2 mt-2"
```
Comment (neekolas) [inline apps/web/src/components/Notification/Type/LikeNotification.tsx:64]: Prettier won't let me commit without making these changes

**Comment by neekolas** [comment]
@bigint resolved the conflicts

**Comment by bigint** [comment]
Screenshot 2023-04-29 at 1 15 40 PM

@neekolas CI failes 😕

**Comment by neekolas** [comment]
> CI failed 😕

@bigint something seems off about the prettier config used in CI vs local. On my machine it forces me to make those changes to the class ordering. I can manually fix, and turn off Husky to allow me to commit the changes. But seems like something's broken. This is what I get on my machine.
CleanShot 2023-04-29 at 09 56 18@2x

**Comment by bigint** [comment]
@neekolas still we have some lint errors and conflicts 😖 

I can't edit the files cuz it's in fork 😕

**Pull request merged**